src/lib.rs

```diff
@@ -272,7 +275,7 @@ mod tests {
 
         let haystack = "it is a beautiful day in Cairo";
         let wm = TwoByteWM::new(&needles);
-        let results = wm.find(haystack).collect::<Vec<_>>();
+        let results = wm.find(&haystack).collect::<Vec<_>>();
```

Comment:
This should not change, see comment above.

Also it might be useful to add a test specifically for binary data, to see if that works as expected.

Reply:
I added a binary data test as well.